vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -3295,4 +3349,225 @@ private void handleDataProviderChange(DataProvider<T, ?> dataProvider) {
+     * {@link DropMode#BETWEEN}, {@link DropMode#ON_TOP} or
+     * {@link DropMode#ON_TOP_OR_BETWEEN}, by default all the visible rows can
+     * be dropped over.
+     * <p>
+     * A drop filter function can be used to specify the rows that are available
+     * for dropping over. The function receives an item and should return
+     * {@code true} if the row can be dropped over, {@code false} otherwise.
+     * <p>
+     * <em>NOTE: If the filter conditions depend on a specific row that's
+     * currently being dragged, you might want to have the grid's drop mode
+     * disabled by default and set its value only on drag start to avoid the
+     * small period of time during which the user might be able to drop over
+     * unwanted rows. Once the drop end event occurs, the drop mode can be set
+     * back to {@code null} to keep this consistent.
+     * <p>
+     * <em>NOTE: If the filtering conditions change dynamically, remember to
+     * explicitly invoke {@code getDataProvider().refreshItem(item)} for the
+     * relevant items to get the filters re-run for them.
+     */
+    public void setDropFilter(SerializableFunction<T, Boolean> dropFilter) {
+        Objects.requireNonNull(dropFilter, "Drop filter can not be null");
+        this.dropFilter = dropFilter;
+        getDataCommunicator().reset();
+    }
+
+    /**
+     * Sets the drag filter for this drag source.
+     * <p>
+     * When the {@link #setRowsDraggable(boolean)} has been used to enable
+     * dragging, by default all the visible rows can be dragged.
+     * <p>
+     * A drag filter function can be used to specify the rows that are available
+     * for dragging. The function receives an item and returns {@code true} if
+     * the row can be dragged, {@code false} otherwise.
+     * <p>
+     * <em>NOTE: If the filtering conditions change dynamically, remember to
+     * explicitly invoke {@code getDataProvider().refreshItem(item)} for the
+     * relevant items to get the filters re-run for them.
+     */
+    public void setDragFilter(SerializableFunction<T, Boolean> dragFilter) {
+        Objects.requireNonNull(dragFilter, "Drag filter can not be null");
+        this.dragFilter = dragFilter;
+        getDataCommunicator().reset();
+    }
+
+    /**
+     * Sets a generator function for customizing drag data. The generated value
+     * will be accessible using the same {@code type} as the generator is set
+     * here. The function is executed for each item in the Grid during data
+     * generation. Return a {@link String} to be appended to the row as {@code
+     * type} data.
+     * <p>
+     * Example, building a JSON object that contains the item's values:
+     *
+     * <pre>
+     *     dragSourceExtension.setDragDataGenerator("application/json", item ->
+     * {
+     *         StringBuilder builder = new StringBuilder();
+     *         builder.append("{");
+     *         getParent().getColumns().forEach(column -> {
```

Comment:
Who is `getParent()`? What is the context?

Reply:
Uh, this was very bad copy-paste. Removed the malfunctioning example.